store/tikv/gc_worker.go

```diff
@@ -43,10 +44,37 @@ type GCWorker struct {
 	lastFinish  time.Time
 	cancel      goctx.CancelFunc
 	done        chan error
+
+	session tidb.Session
+
+	spsession   tidb.Session
+	sessionLock sync.Mutex
+}
+
+// StartSafePointChecker triggers SafePoint Checker on each tidb
+func (w *GCWorker) StartSafePointChecker() {
+	w.spsession = createSession(w.store)
+	go func() {
+		for {
+			select {
+			case <-w.store.spMsg:
+				return
+			default:
+				spCachedTime := time.Now()
+				cachedSafePoint, err := w.loadSafePoint(gcSavedSafePoint)
+				if err == nil {
+					w.store.UpdateSPCache(cachedSafePoint, spCachedTime)
+				} else {
+					log.Warnf("There is no safepoint or the read fails")
```

Comment:
Those two cases is different, I think just read fails  should return error,
no safepoint equals that safepoint = 0
You should distinguish them and print error information.

Reply:
Thanks, I have revised it.